File: WinPaletter/GUI/Elements/Others/ToolStripMenuItem.vb
Imports System.ComponentModel

Namespace UI.WP

    <Description("ToolStripMenuItem fixed To respect dark/light mode")>
    Public Class ToolStripMenuItem : Inherits System.Windows.Forms.ToolStripMenuItem
        Protected Overrides Sub OnPaint(e As PaintEventArgs)
            MyBase.OnPaint(e)
            e.Graphics.TextRenderingHint = My.RenderingHint
            e.Graphics.Clear(BackColor)
            Using br As New SolidBrush(ForeColor) : e.Graphics.DrawString(Text, Font, br, New Rectangle(0, 0, Width, Height), TextAlign.ToStringFormat) : End Using
        End Sub
    End Class

End Namespace

File: WinPaletter/GUI/Elements/Others/ToolStripMenuItem.cs
using System.ComponentModel;
using System.Drawing;
using System.Drawing.Text;
using System.Windows.Forms;

namespace WinPaletter.UI.WP
{
    [Description("ToolStripMenuItem fixed to respect dark/light mode")]
    public class ToolStripMenuItem : System.Windows.Forms.ToolStripMenuItem
    {
        public ToolStripMenuItem()
        {

        }

        protected override void OnPaint(PaintEventArgs e)
        {
            Graphics G = e.Graphics;

            G.TextRenderingHint = DesignMode ? TextRenderingHint.ClearTypeGridFit : Program.Style.TextRenderingHint;

            if (Enabled && Selected)
            {
                G.FillRectangle(SystemBrushes.MenuHighlight, new Rectangle(1, 1, Width - 2, Height - 2));
            }
            else
            {
                using (SolidBrush br = new(BackColor))
                {
                    G.FillRectangle(br, new Rectangle(1, 1, Width - 2, Height - 2));
                }
            }

            using (StringFormat sf = base.TextAlign.ToStringFormat())
            {
                using (SolidBrush br = new(ForeColor))
                {
                    G.DrawString(Text, Font, br, new Rectangle(1, 1, Width - 2, Height - 2), sf);
                }
            }


        }
    }
}